FirebaseDynamicLinks/Sources/Utilities/FDLUtilities.m

```diff
@@ -204,64 +204,92 @@ BOOL FIRDLOSVersionSupported(NSString *_Nullable systemVersion, NSString *minSup
 BOOL FIRDLIsURLForAllowedCustomDomain(NSURL *_Nullable URL) {
   BOOL customDomainMatchFound = false;
   for (NSURL *allowedCustomDomain in FIRDLCustomDomains) {
-    // All custom domain host names should match at a minimum.
+    // At least one custom domain host name should match at a minimum.
     if ([allowedCustomDomain.host isEqualToString:URL.host]) {
       NSString *urlStr = URL.absoluteString;
       NSString *domainURIPrefixStr = allowedCustomDomain.absoluteString;
 
       // Next, do a string compare to check if entire domainURIPrefix matches as well.
-      if (([URL.absoluteString rangeOfString:allowedCustomDomain.absoluteString
-                                     options:NSCaseInsensitiveSearch | NSAnchoredSearch]
+      if (([urlStr rangeOfString:domainURIPrefixStr
+                         options:NSCaseInsensitiveSearch | NSAnchoredSearch]
                .location) == 0) {
-        // The (short) URL needs to be longer than the domainURIPrefix, it's first character after
-        // the domainURIPrefix needs to be '/' and should be followed by at-least one more
-        // character.
-        if (urlStr.length > domainURIPrefixStr.length + 1 &&
-            ([urlStr characterAtIndex:domainURIPrefixStr.length] == '/')) {
-          // Check if there are any more '/' after the first '/'trailing the
-          // domainURIPrefix. This does not apply to unique match links copied from the clipboard.
-          // The clipboard links will have '?link=' after the domainURIPrefix.
-          NSString *urlWithoutDomainURIPrefix =
-              [urlStr substringFromIndex:domainURIPrefixStr.length + 1];
-          if ([urlWithoutDomainURIPrefix rangeOfString:@"/"].location == NSNotFound ||
-              [urlWithoutDomainURIPrefix rangeOfString:@"?link="].location != NSNotFound) {
-            customDomainMatchFound = true;
-            break;
-          }
+        NSString *urlWithoutDomainURIPrefix = [urlStr substringFromIndex:domainURIPrefixStr.length];
+        BOOL matchesRegularExpression =
+            ([urlWithoutDomainURIPrefix
+                 rangeOfString:@"((\\/[A-Za-z0-9]+)|((\\?|\\/\\?)link=https?.*))"
```

Comment:
Can you add a comment about what we're looking for here? Is this any different than the above code? If so, why?

Reply:
The changes added here are: 
1. Moved the logic to Regex.
2. Old code wont consider any number of paths after the whitelisted custom domain as valid.
3. Added check to make sure the link query param starts with http/https
4. Previous code was expecting the presence of either one path after the custom domain prefix or a link query param which is not the case for a valid custom domain.

Reply:
Usually this code execution flow will happens only once after the app is started ie. when the app is opened through universal link, so I believe its better to not cache it since it would stay in the memory even though we dont need it further. LMK if you think otherwise.

Reply:
SGTM

Reply:
> 3. Added check to make sure the link query param starts with http/https

Re: 3, would a link query param work without the http/https before and now it won't? Or would it fail in a different way (silently or not) previously and now we're failing in a more sensible way? Just want to make sure we're not breaking any existing setups.

Reply:
No...previously, the link param without http/https will bypass client validation and will gets rejected from the backend. Now with this change, we will filter it from the client itself.

Reply:
As discussed over GVC with @ryanwilson , Upon checking it is found that even though the server response for this link is un successful, the client SDK was locally creating a dynamic link object and passing to the SDK users. This is going to be breaking with this change as the old behavior is un intended.